### PR TITLE
Try to add function

### PR DESCRIPTION
**Pull request by wangzhen11aaa**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

This is a draft PR. I have some questions before coding.

I  found that all the function register into the map (FactoryFuncRef type), the second  function parameter has pub type FactoryFunc = fn(name: &str) -> Result<Box<dyn Function>>; 

A lot of toInt8/16/32/64 function,  I could create a file per function like  all the functions registered in map, in that way,  a world of duplicated codes would show, that's bad. 

Why did we design the function registered into the map  has a display_name?

## Changelog
add toint8/16/32/64 .. functions.

## Related Issues
https://github.com/datafuselabs/datafuse/issues/967
Fixes #issue
https://github.com/datafuselabs/datafuse/issues/967
## Test Plan
Local Test
Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by BohuTANG** [comment]
Hi @wangzhen11aaa 
display_name used to show the function name consistent with the user side, for example:
select SLeeP(2);
SLeeP is the display name here.

**Comment by wangzhen11aaa** [comment]
> Hi @wangzhen11aaa
> display_name used to show the function name consistent with the user side, for example:
> select SLeeP(2);
> SLeeP is the display name here.

Thanks.

**Comment by wangzhen11aaa** [comment]
In this pull request, added the toint8 function.
Next, I want to  add  the remaining functions  like the ComparisionFunction pattern and  put all the cast function into one file(or mod). 
And add some unit tests about these functions.

**Comment by sundy-li** [comment]
Since we already have FunctionCast, why not just register other cast aliases? 
This is the simplest way to work.

```
toUInt8(col) ---> Cast(col as UInt8)
```

**Comment by wangzhen11aaa** [comment]
> Since we already have FunctionCast, why not just register other cast aliases?
> This is the simplest way to work.
> 
> ```
> toUInt8(col) ---> Cast(col as UInt8)
> ```

First version. I will try that method later.

**Comment by wangzhen11aaa** [comment]
> Since we already have FunctionCast, why not just register other cast aliases?
> This is the simplest way to work.
> 
> ```
> toUInt8(col) ---> Cast(col as UInt8)
> ```

I want to create a new mod named cast_function_template.rs.
codes like this.
"""
   pub struct CastFunctionTemplate;

impl CastFunctionTemplate{
    pub fn try_create_func(display_name: &str) -> Result<Box<dyn Function>> {
        match display_name{
            Some("int8") => CastFunction(DateType::Int8),

        }
    }
}

"""
The display_name will be a problem. I think.

**Comment by sundy-li** [comment]
See :
```
HashesFunction::register
Expression  column_name
```

File: common/functions/src/scalars/expressions/cast_function_template.rs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+use common_datavalues::DataType;
+use common_exception::ErrorCode;
+use common_exception::Result;
+
+use crate::scalars::CastFunction;
+use crate::scalars::Function;
+
+pub struct CastFunctionTemplate;
+
+impl CastFunctionTemplate {
```
Comment (sundy-li) [inline common/functions/src/scalars/expressions/cast_function_template.rs:10]: Why not put these into registers.

Comment (wangzhen11aaa) [reply common/functions/src/scalars/expressions/cast_function_template.rs]: Don't want to change original codes that not familiar with. :)

File: common/functions/src/scalars/expressions/expression.rs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+use crate::scalars::FactoryFuncRef;
+
+#[derive(Clone)]
+pub struct ToCastFunction;
+
+impl ToCastFunction {
+    pub fn register(map: FactoryFuncRef) -> Result<()> {
+        let mut map = map.write();
+        map.insert("toint8", |display_name| {
+            CastFunction::create(display_name.to_string(), DataType::Int8)
```
Comment (sundy-li) [inline common/functions/src/scalars/expressions/expression.rs:17]: All is Int8 ?

File: common/functions/src/scalars/expressions/cast_test.rs [hunk 3]
```diff
@@ -38,7 +38,7 @@ fn test_cast_function() -> Result<()> {
             display: "CAST",
             nullable: false,
             columns: vec![Series::new(vec!["20210305", "20211024"]).into()],
-            func: CastFunction::create(DataType::Int32),
+            func: CastFunction::create("cast".to_string(), DataType::Int32),
             cast_type: DataType::Date32,
             expect: Series::new(vec![20210305i32, 20211024]),
```
Comment (sundy-li) [inline common/functions/src/scalars/expressions/cast_test.rs:43]: expect type is i32, but cast type is Date32

File: common/functions/src/scalars/expressions/cast_test.rs [hunk 4]
```diff
@@ -47,11 +47,10 @@ fn test_cast_function() -> Result<()> {
...
     for t in tests {
         let rows = t.columns[0].len();
-        let func = t.func;
+        let func = t.func.unwrap();
         if let Err(e) = func.eval(&t.columns, rows) {
             assert_eq!(t.error, e.to_string());
         }
-
         // Display check.
         let expect_display = t.display.to_string();
```
Comment (sundy-li) [inline common/functions/src/scalars/expressions/cast_test.rs:55]: need to test the return_type to match the expected data_type

Comment (sundy-li) [reply common/functions/src/scalars/expressions/cast_test.rs:55]: And stateless test is needed.

File: common/functions/src/scalars/expressions/cast_test.rs (hunk 3 above)
Comment (wangzhen11aaa) [reply common/functions/src/scalars/expressions/cast_test.rs:43]: Ok, The original code is  DateType::Int32..
I will change it into Date32.

Comment (wangzhen11aaa) [reply common/functions/src/scalars/expressions/cast_test.rs:43]: > expect type is i32, but cast type is Date32

DateType is 2020-03-05 not 20200305. 
The original code is Int32.. And it passed.

File: common/functions/src/scalars/expressions/cast_test.rs (hunk 4 above)
Comment (wangzhen11aaa) [reply common/functions/src/scalars/expressions/cast_test.rs:55]: All stateless test is passed.

**Review by sundy-li — APPROVED** [review]
LGTM

**Comment by BohuTANG** [comment]
Thank you @wangzhen11aaa

**Pull request merged**